### ISSUE TITLE
Optional dependencies required to pass nosetests

### ISSUE THREAD

**Issue by jpober** [RadioAstronomySoftwareGroup/pyuvdata#451]
healpy and h5py are both listed as optional dependencies, but nosetests returns errors if they're not installed (rather than skips the tests, which [I think] is the behavior for tests which require casacore).

**Comment by bhazelton** [comment]
Yes. I think the reason for skipping the casa tests was because it seemed burdensome to have to install casacore. But we do call those packages optional. So we either need to make those tests skip or we need to document that they're required for testing (as is nose which is not a normal dependency). Which approach would you vote for?

**Comment by jpober** [comment]
My feelings are not strong on this one.  I suppose keeping the test-related code out of the main class seems preferable, but really only marginally so.
